onnxscript/function_libs/torch_lib/graph_building.py

```diff
@@ -447,7 +447,9 @@ def add_input(
         return tensor_value  # type: ignore[return-value]
 
     @runtime_typing.checked
-    def add_initializer(self, name: str, value: torch.Tensor) -> TorchScriptTensor:
+    def add_initializer(
+        self, name: str, value: torch.Tensor, save_value_to_default: bool = True
```

Comment:
I feel the flag `save_value_to_default` is same as making `value` an `Optional[torch.Tensor]` and store to map when not None.

Both feels very weird, and are doing things like "add input but also help me construct the symbolic connection as if this is an initializer, except that there is no initializer data". It really doesn't make sense until at the very end fake mode serialization comes in and brings back all the initializer data and reveals "hey that's why". The whole process also assumes all missing gap will be precisely filled by fake mode serialization for it to work.

We need to be careful with API changes as it will introduce backward compatibility dependency: we cannot remove this arg even if later exporter refactored since people using torch 2.1 with newer onnxscript would crash otherwise. 

Maybe after all introducing fake tensor concept into graph building is the best solution for now ... We do a filter in `to_model_proto` and only serialize initializer if tensor is concrete.